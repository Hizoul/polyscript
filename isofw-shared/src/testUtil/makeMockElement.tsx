import * as React from "react"

const makeMockElement = (name: string, showChildren: boolean = false) => {
  return (props: any) => (
    <div className={name}>
      {JSON.stringify({...props, children: undefined}, undefined, 2)}
      {props.children}
    </div>
  )
  }

export default makeMockElement
