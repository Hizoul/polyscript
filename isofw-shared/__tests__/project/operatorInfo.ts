import makeMockElement from "isofw-shared/src/testUtil/makeMockElement"
import operatorInfoTest from "isofw-shared/src/tests/project/operatorInfo"
import SharedOperatorInfo from "isofw-shared/src/components/project/operatorInfo";

operatorInfoTest(SharedOperatorInfo(makeMockElement("operatorinfo")))
