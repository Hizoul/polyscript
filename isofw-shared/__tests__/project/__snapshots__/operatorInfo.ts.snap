// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` operator info test  works as planned:  creation of cameras  1`] = `
Array [
  Object {
    "_id": "c19999999999999999999999",
    "ip": "localhost:4202/c1",
    "name": "Camera#1",
  },
  Object {
    "_id": "c29999999999999999999999",
    "ip": "localhost:4202/c2",
    "name": "Camera#2",
  },
  Object {
    "_id": "c39999999999999999999999",
    "ip": "localhost:4202/c3",
    "name": "Camera#3",
  },
]
`;

exports[` operator info test  works as planned:  creation of proInjects  1`] = `
Array [
  Object {
    "_id": "b19999999999999999999999",
    "cameras": Array [
      "c19999999999999999999999",
      "c29999999999999999999999",
      "c39999999999999999999999",
    ],
    "name": "pn1",
    "operatorCameraMapping": Array [
      Object {
        "cameras": Array [
          "c19999999999999999999999",
          "c39999999999999999999999",
        ],
        "operator": "u1",
      },
      Object {
        "cameras": Array [
          "c29999999999999999999999",
        ],
        "operator": "u2",
      },
    ],
    "operators": Array [
      "u1",
      "u2",
    ],
    "shot": 1,
  },
  Object {
    "_id": "b29999999999999999999999",
    "cameras": Array [
      "c19999999999999999999999",
      "c29999999999999999999999",
    ],
    "name": "pn2",
    "operatorCameraMapping": Array [
      Object {
        "cameras": Array [
          "c19999999999999999999999",
        ],
        "operator": "u3",
      },
      Object {
        "cameras": Array [
          "c29999999999999999999999",
        ],
        "operator": "u1",
      },
    ],
    "operators": Array [
      "u1",
      "u3",
    ],
    "shot": 8,
  },
  Object {
    "_id": "b39999999999999999999999",
    "cameras": Array [
      "c19999999999999999999999",
      "c39999999999999999999999",
    ],
    "name": "pn3",
    "operatorCameraMapping": Array [
      Object {
        "cameras": Array [
          "c19999999999999999999999",
        ],
        "operator": "u3",
      },
      Object {
        "cameras": Array [
          "c39999999999999999999999",
        ],
        "operator": "u2",
      },
    ],
    "operators": Array [
      "u2",
      "u3",
    ],
    "shot": 6,
  },
]
`;

exports[` operator info test  works as planned: Before anything 1`] = `
<div
  className="operatorinfo"
>
  {
  "item": {
    "_id": "b19999999999999999999999",
    "name": "pn1",
    "shot": 1,
    "cameras": [
      "c19999999999999999999999",
      "c29999999999999999999999",
      "c39999999999999999999999"
    ],
    "operators": [
      "u1",
      "u2"
    ],
    "operatorCameraMapping": [
      {
        "operator": "u1",
        "cameras": [
          "c19999999999999999999999",
          "c39999999999999999999999"
        ]
      },
      {
        "operator": "u2",
        "cameras": [
          "c29999999999999999999999"
        ]
      }
    ]
  },
  "currentCameras": [],
  "filteredList": []
}
</div>
`;

exports[` operator info test  works as planned: Invalid user selected 1`] = `
<div
  className="operatorinfo"
>
  {
  "item": {
    "_id": "b19999999999999999999999",
    "name": "pn1",
    "shot": 1,
    "cameras": [
      "c19999999999999999999999",
      "c29999999999999999999999",
      "c39999999999999999999999"
    ],
    "operators": [
      "u1",
      "u2"
    ],
    "operatorCameraMapping": [
      {
        "operator": "u1",
        "cameras": [
          "c19999999999999999999999",
          "c39999999999999999999999"
        ]
      },
      {
        "operator": "u2",
        "cameras": [
          "c29999999999999999999999"
        ]
      }
    ]
  },
  "currentOperator": "blabliblu",
  "currentCameras": [],
  "filteredList": []
}
</div>
`;

exports[` operator info test  works as planned: Removed user selection 1`] = `
<div
  className="operatorinfo"
>
  {
  "item": {
    "_id": "b19999999999999999999999",
    "name": "pn1",
    "shot": 1,
    "cameras": [
      "c19999999999999999999999",
      "c29999999999999999999999",
      "c39999999999999999999999"
    ],
    "operators": [
      "u1",
      "u2"
    ],
    "operatorCameraMapping": [
      {
        "operator": "u1",
        "cameras": [
          "c19999999999999999999999",
          "c39999999999999999999999"
        ]
      },
      {
        "operator": "u2",
        "cameras": [
          "c29999999999999999999999"
        ]
      }
    ]
  },
  "currentOperator": "",
  "currentCameras": [],
  "filteredList": []
}
</div>
`;

exports[` operator info test  works as planned: first user selected 1`] = `
<div
  className="operatorinfo"
>
  {
  "item": {
    "_id": "b19999999999999999999999",
    "name": "pn1",
    "shot": 1,
    "cameras": [
      "c19999999999999999999999",
      "c29999999999999999999999",
      "c39999999999999999999999"
    ],
    "operators": [
      "u1",
      "u2"
    ],
    "operatorCameraMapping": [
      {
        "operator": "u1",
        "cameras": [
          "c19999999999999999999999",
          "c39999999999999999999999"
        ]
      },
      {
        "operator": "u2",
        "cameras": [
          "c29999999999999999999999"
        ]
      }
    ]
  },
  "currentOperator": "u1",
  "currentCameras": [
    "c19999999999999999999999",
    "c39999999999999999999999"
  ],
  "filteredList": []
}
</div>
`;
